File: fsharp/darts/Darts.fs
module Darts

    open System

    let private inner:double = 1.0

    let private middle:double = 5.0

    let private outer:double = 10.0

    let private radius (x:double) (y:double):double =
        (x * x + y * y)
        |> Math.Sqrt

    let private (|GreaterThan|_|) (a:'a) (b:'a) =
        if a < b then
            Some()
        else
            None

    let score (x:double) (y:double):int =
        let distance:double = radius x y
        match distance with
        | GreaterThan outer -> 0
        | GreaterThan middle -> 1
        | GreaterThan inner -> 5
        | _ -> 10

File: solutions/scala/darts/1/src/main/scala/Darts.scala
object Darts
{
    private def inner:Double = 1.0

    private def middle:Double = 5.0

    private def outer:Double = 10.0

    private def radius (x:Double, y:Double):Double =
        Math.sqrt(x * x + y * y)

    def score(x:Double, y:Double):Int =
    {
        def distance:Double = radius(x, y)
        distance match
        case distance if distance > outer => 0
        case distance if distance > middle => 1
        case distance if distance > inner => 5
        case _ => 10
    }
}